db/dbProvider.js

```diff
@@ -0,0 +1,20 @@
+class DBProvider {
+    constructor({ pool }) {
+        this.pool = pool;
+    }
+
+    async execute(query) {
+        const client = await this.pool.connect();
+        const result = await client.query(query);
+        //await this.pool.end();
+
+        // now with this thing the app is running for some time after it executed query(-ies) (around 5 secs)
+        // but it seems better that pool.end since it doesn't let execute several queries at a time
+
+        await client.release();
+        return result.rows;
```

Comment:
You should think of using this https://medium.com/swlh/node-js-and-singleton-pattern-7b08d11c726a
In general, there is an opinion, that Singleton pattern is kinda "anti-pattern" but it is really helpful for such cases. 
It is obligatory (for now) to use it.